sqlmesh/core/engine_adapter/bigquery.py

```diff
@@ -620,3 +633,51 @@ def should_retry(self, error: t.Type[Exception]) -> bool:
             )
             return True
         return False
+
+
+def select_partitions_expr(
+    schema: str,
+    table_name: str,
+    data_type: t.Union[str, exp.DataType],
+    granularity: t.Optional[str] = None,
+    agg_func: str = "MAX",
+    database: t.Optional[str] = None,
+) -> str:
+    """Generates a SQL expression that aggregates partition values for a table.
+
+    Args:
+        schema: The schema (BigQueyr dataset) of the table.
+        table_name: The name of the table.
+        data_type: The data type of the partition column.
+        granularity: The granularity of the partition. Supported values are: 'day', 'month', 'year' and 'hour'.
+        agg_func: The aggregation function to use.
+        database: The database (BigQuery project ID) of the table.
+
+    Returns:
+        A SELECT statement that aggregates partition values for a table.
+    """
+    partitions_table_name = f"`{schema}`.INFORMATION_SCHEMA.PARTITIONS"
```

Comment:
exp.table(table_name, db=schema, catalog=database)

Reply:
If I do `exp.table_("INFORMATION_SCHEMA.PARTITIONS", db=schema, catalog=database)` the quoting gets messed up: 
```
FROM test_schema.`INFORMATION_SCHEMA.PARTITIONS`
```

Reply:
ah i know why